x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/RollupShardIndexer.java

```diff
@@ -383,18 +385,18 @@ private class RollupBucketBuilder {
         private int tsidOrd = -1;
         private long timestamp;
         private int docCount;
-        private final List<AbstractDownsampleFieldProducer> rollupFieldProducers;
-        private final List<DownsampleFieldSerializer> groupedProducers;
+        private final AbstractDownsampleFieldProducer[] rollupFieldProducers;
+        private final DownsampleFieldSerializer[] groupedProducers;
 
-        RollupBucketBuilder(List<AbstractDownsampleFieldProducer> rollupFieldProducers) {
+        RollupBucketBuilder(AbstractDownsampleFieldProducer[] rollupFieldProducers) {
             this.rollupFieldProducers = rollupFieldProducers;
             /*
              * The rollup field producers for aggregate_metric_double all share the same name (this is
              * the name they will be serialized in the target index). We group all field producers by
              * name. If grouping yields multiple rollup field producers, we delegate serialization to
              * the AggregateMetricFieldSerializer class.
              */
-            groupedProducers = rollupFieldProducers.stream()
+            groupedProducers = Arrays.stream(rollupFieldProducers)
```

Comment:
Same here...maybe a simple for loop has better performance.

Reply:
I think we can keep the stream usage here and in `TimeSeriesBucketCollector` constructor here. 
This is called once per shard level downsample execution, so it isn't hot code, like in the places I made changes in this PR.